lib/os_places_api/client.rb

```diff
@@ -57,19 +57,25 @@ def filtered_locations(results)
         .map { |result_hash| LocationBuilder.new(result_hash).build_location }
     end
 
+    def unfiltered_locations(results)
+      results.map { |result| result[result.keys.first] } # first key is either "LPI" or "DPA"
+        .uniq { |result_hash| result_hash["UPRN"] }
```

Comment:
There feels like a refactor opportunity here - this method is identical to `filtered_locations` in the setup and in the final `map` call, all that's different is a bit of extra filtering. Perhaps something like:

```ruby
def unfiltered_locations(results, filter = nil)
  results_hash = results.map { |result| result[result.keys.first] } # first key is either "LPI" or "DPA"
    .uniq { |result_hash| result_hash["UPRN"] }
  yield(results_hash) if block_given?
  results_hash.map { |result_hash| LocationBuilder.new(result_hash).build_location }
end

def filtered_locations(results)
  unfiltered_locations(results) do |results_hash|
    results_hash.reject { |result_hash| NATIONAL_AUTHORITIES.include? result_hash["LOCAL_CUSTODIAN_CODE_DESCRIPTION"] }
  end
end
```

Or perhaps something less complex:

```ruby
def unfiltered_locations(results)
  hash_to_locations(results_hash(results))
end

def filtered_locations(results)
  filtered_hash = results_hash(results).reject { |result_hash| NATIONAL_AUTHORITIES.include? result_hash["LOCAL_CUSTODIAN_CODE_DESCRIPTION"] }
  hash_to_locations(filtered_hash)
end
```

(Sorry if I got carried away, I don't get to write Ruby much these days! 💎)